src/plugins/intel_gpu/src/graph/graph_optimizer/compile_graph.cpp

```diff
@@ -90,7 +90,8 @@ void compile_graph::run(program& p) {
 
         // TODO: need to come up with better handling of unsupported shape agnostic cases
         // e.g. process exceptions from choose_impl() and ignore those for dynamic parameters
-        if (node->is_type<fully_connected>() && node->is_dynamic() && node->get_output_pshape().size() > 3)
+        if (node->is_type<fully_connected>() && node->is_dynamic()
```

Comment:
Since no onednn primitives support shape agnostics now, frankly we don't need to check FC condition.

How about add additional if states for setting can_select_iml to false when all onednn primitive with dynamic?

This could be more clear.

Reply:
I think we can just revert this change to original impl. 
Because no shape_agnostic_kernel is now used for onednn FC. 
Still origianl control for get_output_pshae().size() > 3 should be remained.

Reply:
Fixed it.